### PR TITLE
fix(graphql): index rds query

### PR DESCRIPTION
**Pull request by sundersc**
- Fixes the RDS Index query request mapping template.
- Fixes the RDS lambda to process index query.

### CODE REVIEW

**Review by marcvberg — APPROVED** [review]
LGTM

File: packages/amplify-graphql-model-transformer/rds-lambda/interfaces/BaseRequest.ts [hunk 1]
```diff
@@ -9,14 +9,14 @@ export interface RequestArgs extends BaseRequestArgs {
 
 export interface BaseRequest {
   table: string;
-  operation: 'GET' | 'LIST' | 'CREATE' | 'UPDATE' | 'DELETE';
+  operation: 'GET' | 'LIST' | 'CREATE' | 'UPDATE' | 'DELETE' | 'INDEX';
```
Comment (phani-srikar) [inline packages/amplify-graphql-model-transformer/rds-lambda/interfaces/BaseRequest.ts:12]: `INDEX` doesn't feel like it's an operation. It's more like part of DDL.

File: packages/amplify-graphql-model-transformer/rds-lambda/clients/mysql/MySQLClient.ts [hunk 2]
```diff
@@ -89,4 +91,22 @@ export abstract class MySQLClient extends DBClient {
 
     return result ? result[0] : {};
   }
+
+  private executeIndex = async (request: IndexRequest): Promise<any> => {
```
Comment (phani-srikar) [inline packages/amplify-graphql-model-transformer/rds-lambda/clients/mysql/MySQLClient.ts:95]: "Execute Query with Index" is more readable here.

**Review by phani-srikar — APPROVED** [review]
Non-blocking comments.

**Review by alharris-at — APPROVED** [review]

File: packages/amplify-graphql-model-transformer/rds-lambda/interfaces/BaseRequest.ts (hunk 1 above)
Comment (sundersc) [reply packages/amplify-graphql-model-transformer/rds-lambda/interfaces/BaseRequest.ts:12]: I'm going to cut a tagged release after this. I will clean up these method names in the next PR.

**Pull request merged**